jnius/reflect.py

```diff
@@ -232,39 +232,36 @@ def autoclass(clsname):
         constructors.append((sig, constructor.isVarArgs()))
     classDict['__javaconstructor__'] = constructors
 
-    classHierachy=[]
-    identifyHierarchy(c, 0, classHierachy, not c.isInterface())
-    classHierachy.reverse()
-
-    print("autoclass(%s) intf %r hierarchy is %s" % (clsname,c.isInterface(),classHierachy))
-    clsDone=set()
+    class_hierachy = identify_hierarchy(c, 0, not c.isInterface())
+    
+    log.debug("autoclass(%s) intf %r hierarchy is %s" % (clsname,c.isInterface(),class_hierachy))
```

Comment:
that one might be less useful now it's a generator though :sweat_smile: maybe cast class_hierarchy to a list first to be able to both log and then iterate on it in the loop.